bin/xbundle_convert

```diff
@@ -1,69 +1,72 @@
+
+# local
 from xbundle import XBundle, run_tests
 
 # pylint: disable=star-args
 
-def usage():
-    """
-    print usage
-    """
-    print "Usage: python xbundle.py [--force-studio] [cmd] [infn] [outfn]"
-    print "where:"
-    print "  cmd = test:    run unit tests"
-    print "  cmd = convert: convert between xbundle and edX directory format"
-    print "                 the xbundle filename must end with .xml"
-    print "  --force-studio forces <sequential> to always be followed by <vertical> in export"
-    print "                 this makes it compatible with Studio import"
-    print ""
-    print "examples:"
-    print "  python xbundle.py convert ../data/edx4edx edx4edx_xbundle.xml"
-    print "  python xbundle.py convert edx4edx_xbundle.xml ./"
-
 def main():
     """
     Convert between OLX and xbundle XML formats.
     """
-    if len(sys.argv) < 2:
-        usage()
-        sys.exit(0)
+    # This parses the docstring above and derives all arguments.
+    # If the required arguments are missing, it prints basic usage
+    # and exits automatically.
+    # If required arguments are satisfied, it returns a dict.
+    args = docopt(__doc__)
 
-    argc = 1
     options = dict(keep_urls=True)
-    if len(sys.argv) > argc and sys.argv[argc] == '--force-studio':
-        argc += 1
+    if args['--force-studio']:
         options['force_studio_format'] = True
 
-    cmd = sys.argv[argc]
-
-    if cmd == 'test':
+    if args['test']:
         run_tests()
+        return
+    if args['--version']:
+        print "xbundle {0}".format(get_distribution("xbundle").version)
+        return
 
-    elif cmd == 'convert':
-        argc += 1
-        infn = sys.argv[argc]
-        outfn = sys.argv[argc + 1]
-        bundle = XBundle(**options)
-        if infn.endswith('.xml'):
-            print "Converting xbundle file '%s' to edX xml directory '%s'" % (infn, outfn)
-            bundle.load(infn)
-            bundle.export_to_directory(outfn)
-            print "done"
-        elif outfn.endswith('.xml'):
-            print "Converting edX xml directory '%s' to xbundle file '%s'" % (infn, outfn)
-            bundle.import_from_directory(infn)
-            bundle.save(outfn)
-            print "done"
-        else:
-            usage()
+    bundle = XBundle(**options)
+    input_path = args['<input>']
+    output_path = args['<output>']
+    if input_path.endswith('.xml'):
+        print "Converting xbundle '{0}' to edX directory '{1}'".format(
+            input_path, output_path)
+        bundle.load(input_path)
+        bundle.export_to_directory(output_path)
+        print "done"
+    elif output_path.endswith('.xml'):
+        print "Converting edX directory '{0}' to xbundle '{1}'".format(
+            input_path, output_path)
+        bundle.import_from_directory(input_path)
+        bundle.save(output_path)
+        print "done"
     else:
-        usage()
+        print "Invalid input; run with --help for more info."
```

Comment:
Perhaps this should also cause an exit with a status code of `1` here after printing to indicate failure. E.g. `exit(1)`.

This is out of the scope of the PR, but if there are multiple points of failure deeper in the conversion process, we might want to think about having different status codes for different kinds of failures down the line if we'll be programmatically integrating this with other tools.